### DISCUSSION TITLE
Why do I get folders that are different but contain no files of interest.

### DISCUSSION THREAD

**Discussion by CataCluj** [WinMerge/winmerge#1199] [Q&A]
Hi,
I have a very simple filter file like this:
  def: exclude
  f: \.c$ ## Filter for filename
  f: \.h$ ## Filter for filename
  d: \\*$ ## Filter for directory
I do get the .c, .h that changed in main and subfolders but I also get folders that have NO .c or .h file
What's with the .git, Application, ATI.X folders and their subfolders?  I don't care if they are different or even exist only on one side. How do I only get the .c and .h files?
Thanks,
Cat
![WinMerge](https://user-images.githubusercontent.com/33530615/154745711-b847e579-ff25-429f-bd3f-9f39cbbec772.png)
s

**Comment by CataCluj** [comment]
Nobody? Is this abandoned?

**Comment by sdottaka** [comment]
Unfortunately, WinMerge does not allow you to specify that folders that do not contain certain files be excluded.

**Reply by CataCluj** [reply]
Thank you sdottaka,
I don't expect to exclude folders "that do not contain certain files".
I DO expect to exclude folders that have NO files of interest! If I care only about .h and .c files, why do I get folders that contain neither?
I think this needs to be added/changed.
Cat

**Comment by sdottaka** [comment]
You are specifying that you are interested in all folders because you have set the filter as follows: WinMerge is working as per this filter.

~~~
d: \*$ ## Filter for directory
~~~

Perhaps WinMerge needs an option to exclude a folder if all files in that folder were ignored.

**Reply by Gitoffthelawn** [reply]
> Perhaps WinMerge needs an option to exclude a folder if all files in that folder were ignored.

That might be a useful option.

**Comment by CataCluj** [comment]
Should/could I have specified folders differently to get the result I wanted?
I was/am not sure I know totally how to use it; I thought maybe there is a way, but maybe that option is what's needed.
In fact, why make it an option: nobody would care about folders without files of interest, no?  They just clutter the interface. In fact that's my main objection; there could be lots of folders with very few files of interest as you can see in my example.

**Comment by sdottaka** [comment]
In most cases, if there are no files in the folders, you may not need to see the folders. But even if that's not the case now, you might want to make sure that the file doesn't exist in the folder.
Therefore, we cannot change the behavior as you say.

By the way, have you tried disabling tree mode?
Disabling tree mode as shown below may help a little because files and folders can be displayed separately. 
![image](https://user-images.githubusercontent.com/98126/156761275-4ff2bdf6-f037-40f8-88f6-1211379a0a18.png)

**Reply by CataCluj** [reply]
I see how showing the files separately might help some, but I think it's good to know where files are; same filenames might be in different folders so that's a problem. Also, sometimes you know what folder you need to look for things, not the file names.

**Comment by CataCluj** [comment]
1. I would be SURE no files of interest are in ANY folders that were not shown
2. I don't know how I disable Tree Mode but it seems to only flatten how it shows the same number of useless folders. Not useful in this case.
I am curious, how did you get the files to obtain the picture above?  I didn't think I sent any files.

**Reply by sdottaka** [reply]
To disable tree mode, uncheck the Tree Mode menu item as shown below

![image](https://user-images.githubusercontent.com/98126/156853816-7cccb2fd-6aa5-49fd-8821-5cfdd1973d0a.png)

> I am curious, how did you get the files to obtain the picture above? I didn't think I sent any files.

I created it from the screenshot you attached.
